design/specs/record-page.md

```diff
@@ -1,71 +1,137 @@
+- One line in large font size. It contains:
+    - A "record" icon. (E.g. something like [this](https://fontawesome.com/icons/memo-pad?s=solid))
+    - The **Record Summary** string
+    - A dropdown icon that opens a menu to perform the following actions on the record:
+        - Delete
+        - *Potentially other actions in the future, such as "Add field", "Duplicate", etc.*
 
-Once the entire flow is specified, this design is likely to change.
+## Direct Fields
 
-### C.Record Fields 
+- This is a section of the page which comes after the title.
+- It contains a form-like area which has many labeled input fields. One field exists for each column in this table.
+- The fields are presented in the same order that they occur in the Table Page.
+- Field labels display the column name along with the icon used within the column header on the Table Page.
+- Edits to the field values are saved on blur, with a similar UX to editing cell values in a table. There is no save button.
+- Similar to the Table Page, a persistent but subtle status message appears at the top of all the fields which reads "Saving" or "All changes saved" when appropriate.
+- FK fields display the record summary for the linked record and allow the user to navigate to its Record Page.
+- The layout of fields and their associated labels can be improvised during development (and adjusted during future design iterations).
 
-The record fields will be shown as a collection of input controls. As with tables, the input fields may be interacted with by users to modify the data. The fields will be presented in the same format and order as in tables. 
 
-### D.Links to Record and Queries 
+## Widgets
 
-The record page will also have a section with links to the selected record. This work is currently being defined, and it is likely to change. However, the aim is to add a default query for the tables with connected records and present them as tables or visualizations that the user can edit.
+One "widget" allows the user to see (and potentially modify) other data in the same schema *from the context of this record*.
 
-Custom embedded queries created by users with Data Explorer will be used to define the tables' fields and options.
+**Example:** On a Patrons record, we use a widget to see the Checkouts records associated with the patron.
 
-## Navigating to the record page
+*Note: The name "widget" may be a bit awkward, we're open to replacements. A widget is kind of like a query or a view -- but "query" and "view" already mean very specific things and a widget may not necessarily be either a Mathesar query or a Postgres view. Also, although these specs describe a limited set of initial functionality, we expect widgets to gradually evolve into a flexible and powerful feature (more details at the end of these specs), so we've coined a unique (and somewhat generic) name for it.*
```

Comment:
I think "widget" is a fine name, it's used for embedded versions of features quite widely.

For more precision I would call the auto-generated widgets "table widgets" (since they're automatically generated from a table and similar to tables) and the manually-generated widgets "query widgets" (since they're more freeform and are edited using our query builder).

Reply:
Brilliant! This should be addressed in f13e2c75abfd6cba3c3b12e1517426c9bf11dc21